### PR TITLE
Add deploy target

### PR DESCRIPTION
**Pull request by ffjlabo**
**What this PR does**:

Added deployTarget to the model and piped config to use them on the pipedv1.

**Why we need it**:

**Which issue(s) this PR fixes**:

Part of #5252

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/model/application.proto [hunk 1]
```diff
@@ -43,6 +43,8 @@ message Application {
     // This must be one of the provider names registered in the piped.
     // TODO: Add validation for this field.
     string platform_provider = 15;
+
+    repeated string deploy_targets = 16;
```
Comment (Warashi) [inline pkg/model/application.proto:47]: How about adding comments and a validation?
The validation may be like `min_len = 1`

**Review by Warashi — COMMENTED** [review]
almost LGTM
I commented on one nitpick

File: pkg/model/application.proto (hunk 1 above)
Comment (khanhtc1202) [reply pkg/model/application.proto:47]: This should not have validation, or application which created by pipedv0 will be invalided :eyes:

Comment (Warashi) [reply pkg/model/application.proto:47]: Oops, that's right 🙏🏻

Comment (khanhtc1202) [reply pkg/model/application.proto:47]: Also, could you add comment to explain what does this field do or be used for @ffjlabo

**Comment by khanhtc1202** [comment]
@ffjlabo You may need to update dummyApplication to pass the web test 👀

File: pkg/model/application.proto (hunk 1 above)
Comment (ffjlabo) [reply pkg/model/application.proto:47]: Forgot to  add comment 🙏  will do it.

Comment (ffjlabo) [reply pkg/model/application.proto]: @Warashi  @khanhtc1202 fied f31a246

**Comment by ffjlabo** [comment]
@khanhtc1202 fixed for lint/web [3bc49da](https://github.com/pipe-cd/pipecd/pull/5378/commits/3bc49dae6970d88d3635200c5c2970757e80ba65)

**Comment by ffjlabo** [comment]
@khanhtc1202 @Warashi Sorry for my sloppy fixes 🙏 Fixed and passed the CI

**Review by Warashi — APPROVED** [review]
Thanks. LGTM 👍🏻

**Review by khanhtc1202 — APPROVED** [review]
Here you go 👍

**Pull request merged**